bundles/org.eclipse.cdt.lsp.editor.ui/src/org/eclipse/cdt/lsp/editor/ui/preference/LspEditorPreferences.java

```diff
@@ -17,9 +17,12 @@
 import org.eclipse.core.runtime.preferences.PreferenceMetadata;
 
 public final class LspEditorPreferences {
+	public static final String PREFER_LSP_EDITOR = "prefer_lsp_editor";
+	public static final String SERVER_PATH = "server_path";
```

Comment:
These options are currently only taking effect if the preference page is opened (with `performOk` executed) before the server starts up, otherwise server always uses default.

Reply:
That's right. The server settings should be loaded from the preference store. If the preferences are empty, they should be initialized by the server provider values.

Reply:
Sorry, I wasn't clear in my comment. If you startup and don't open preference page, then the preference page settings aren't used at all. AFAIU this is a bug.

Steps:

1. Start eclipse, go to preferences, change server settings
2. Restart Eclipse
3. Open LSP editor

In 3 the default server settings will be used, not the ones entered in 1.

Reply:
I made the test and with the last change I commited it works.